gcip/core/job.py

```diff
@@ -152,18 +152,18 @@ def __init__(
         self._original: Optional[Job]
         """Only set if you get a :meth:`copy()` of this job"""
 
-        if namespace and name:
-            self._name = f"{namespace}-{name}"
-            self._stage = namespace
-        elif namespace:
-            self._name = namespace
-            self._stage = namespace
+        if stage and name:
+            self._name = f"{stage}-{name}"
+            self._stage = stage
+        elif stage:
+            self._name = stage
+            self._stage = stage
         elif name:
             self._name = name
             # default for unset stages is 'test' -> https://docs.gitlab.com/ee/ci/yaml/#stages
             self._stage = "test"
```

Comment:
I know, there is no change here, but if stage `test` is the default, do we need it here?

Reply:
I think for the current implementation of the gcip we need a stage here. Or else - we cannot set no stage. (Maybe) the gcip has a problem, when the stage is set to `None`.